docs/components.md

```diff
@@ -0,0 +1,94 @@
+    - `bash ./<cluster_name>/create_cluster.sh` will result in an error indicating the cluster is already present.
+    - Argument keys are lower case, underscore separated version of the [gcloud container cluster describe](https://cloud.google.com/sdk/gcloud/reference/beta/container/node-pools/describe) command.
+    - If `-f` file is passed in, data are merged with `-D` values ovveriding the file values
+    - If `cluster` is omitted it will act on all clusters in the project
+    - Example: 
+      ```
+      pentagon get gcp.cluster -D project="pentagon" -D zone="us-central1-a -D cluster="pentagon-1" -D get_default_nodepool="true"
+      ```
+
+### gcp.nodepool
+
+- add:
+    - Creates `./<nodepool_name>/create_nodepool.sh` compiled from the data passed in.
+    - `bash ./<nodepool_name>/create_nodepool.sh` will create the nodepool as configured
+    - Argument keys are lower case, underscore separated version of the [gcloud container node-pools create](https://cloud.google.com/sdk/gcloud/reference/beta/container/node-pools/create) command
+    - If a `-f` file is passed in, data are merged with `-D` values ovveriding the file values
+    - Example: 
+      ```
+      pentagon add gcp.nodepool -D name="pentagon-1-nodepool" -D project="pentagon" -D zone="us-central1-a" -D additional_zones="us-central1-b,us-central1-b" -D machine_type="n1-standard-64" --enable-autoscaling
+      ```
+
+- get: 
+    - Creates `./<nodepool_name>/create_nodepool.sh` by querying the state of an existing cluster nodepool and parsing values. For when you have an existing cluster that you want to capture its configuration.
+    - Creates `./<nodepool_name>/create_nodepool.sh` 
+    - `bash ./<nodepool_name>/create_nodepool.sh` will result in an error indicating the cluster is already present.
+    - Argument keys are lower case, underscore separated version of the [gcloud container node-pools describe](https://cloud.google.com/sdk/gcloud/reference/beta/container/node-pools/describe) command
+    - If a `-f` file is passed in, data are merged with `-D` values ovveriding the file values
+    - If `name` is omitted it will act on all nodepool in the cluster
+    - Example:
+        ```
+        pentagon get gcp.nodepool -D project="pentagon" -D zone="us-central1-a -D cluster="pentagon-1" -D name="pentagon-1-nodepool"
+        ```
+
+
+### vpc
+
+- add: 
+    - Creates `./vpc/` directory with Terraform code for the Pentagon default AWS VPC described [here](#network).
+    - `cd ./vpc; make all` will create the vpc as describe by the arguments passed in
+    - In the normal course of using Pentagon and the infrastructure repository, it is unlikely you'll use this component.
+    - Required Arguments:
+        - vpc_name
+        - vpc_cidr_base
+        - aws_availabilty_zones
+        - aws_availability_zone_count
+        - aws_region
+        - infrastructure_bucket
```

Comment:
Do these need to be added to `_required_parameters` for the `Vpc` class in order to enforce this?

Reply:
I opted to not strictly enforce these requirements because there may be a case where you want to run the component and edit the files after the fact.

Reply:
Mind writing something that conveys that then? If there are defaults provided somehow they aren't really required.